x-pack/plugin/search-business-rules/src/yamlRestTest/resources/rest-api-spec/test/search-business-rules/10_pinned_retriever.yml

```diff
@@ -0,0 +1,406 @@
+  - skip:
+      features: headers
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+        explain: true
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc2 }
+  - match: { hits.hits.2._id: doc3 }
+  - match: { hits.hits.0._score: 1.7014124E38 }
+  - match: { hits.hits.1._score: 1.7014122E38 }
+  - match: { hits.hits.2._score: 1.0 }
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1", "doc3"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+        explain: true
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.1._id: doc3 }
+  - match: { hits.hits.2._id: doc2 }
+  - match: { hits.hits.3._id: doc4 }
+  - match: { hits.hits.4._id: doc5 }
+  - match: { hits.hits.0._score: 1.7014124E38 }
+  - match: { hits.hits.1._score: 1.7014122E38 }
+  - match: { hits.hits.2._score: 1.0 }
+
+---
+"pinned retriever combined with rrf":
+  - skip:
+      features: headers
+
+  - do:
+      headers:
+        Content-Type: application/json
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              ids: ["doc1"]
+              retriever:
+                rrf:
+                  retrievers: [
+                    {
+                      standard: {
+                        query: {
+                          term: { text: "document" }
+                        }
+                      }
+                    },
+                    {
+                      standard: {
+                        query: {
+                          term: { text: "three" }
+                        }
+                      }
+                    }
+                  ]
+              rank_window_size: 10
+
+  - match: { hits.total.value: 5 }
+  - match: { hits.hits.0._id: doc1 }
+  - match: { hits.hits.0._score: 1.7014122E38 }
+  - match: { hits.hits.1._id: doc3 }
+  - match: { hits.hits.2._id: doc2 }
+  - match: { hits.hits.1._score < hits.hits.0._score }
+
+---
+"pinned retriever with pagination":
+  - do:
+      search:
+        index: test-index1
+        body:
+          size: 1
+          from: 1
+          retriever:
+            pinned:
+              ids: ["doc1", "doc2"]
+              retriever:
+                standard:
+                  query:
+                    match_all: {}
+
+  - match: { hits.total.value: 5 }
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._id: doc2 }
+
+---
+"pinned retriever as a sub-retriever":
```

Comment:
This will work but not return the expected results. If you're going to test this case, I would expect that you'd get out of ordered results as RRF doesn't respect pinned documents. If we're going to include this test case let's iterate on it.